src/Controller/ArticleController.php

```diff
@@ -75,15 +75,20 @@ public function add(): string
         $categoryManager = new CategoryManager();
         $categories = $categoryManager->getAllCategory();
 
-        $this->uploadFile();
+        if ($_POST) {
+            $newArticle = $_POST;
+            $imageArticle = $this->uploadFile();
+            $newArticle['imgSrc'] = $imageArticle;
 
+            (new ArticleManager())->createArticle($newArticle);
+        }
         return $this->twig->render('Admin/Article/add.html.twig', [
             'authors' => $authors,
             'categories' => $categories
         ]);
     }
 
-    public function uploadFile(): void
+    public function uploadFile(): string
```

Comment:
maybe declare it protected in the interface and here since you never use the uploadfile outside the class ?

Reply:
My bad all interfaces methods MUST be public! Your code is fine